### DISCUSSION TITLE
Can I have ECS and Threadpool without GUI?

### DISCUSSION THREAD

**Discussion by Wandalen** [bevyengine/bevy#9999] [Q&A] [answered]
Hello! I appreciate the fantastic open-source work you've done!

I've managed to isolate the necessary plugins to run my application without a window. However, in this configuration, the Update systems are only called once. To have them executed on each frame, I've found that I need to include the WindowPlugin, which is something I'd like to avoid. Is there a way to have recurring Update calls without adding all the plugins that I don't require in my specific case?

```rust
    .add_plugins( bevy::core::TaskPoolPlugin::default() )
    .add_plugins( bevy::time::TimePlugin::default() )
    .add_plugins( bevy::core::FrameCountPlugin::default() )
    .add_plugins( bevy::input::InputPlugin::default() )
    .add_plugins( bevy::window::WindowPlugin::default() )
    .add_plugins( bevy::a11y::AccessibilityPlugin )
```

**Comment by hymm** [comment]
You can use `ScheduleRunnerPlugin::run_loop(x)`. https://docs.rs/bevy/latest/bevy/app/struct.ScheduleRunnerPlugin.html

**Comment by ItsDoot** [comment, marked as answer]
The [`MinimalPlugins`](https://docs.rs/bevy/latest/bevy/struct.MinimalPlugins.html) PluginGroup provides you with a headless runtime, which includes a schedule runner via the [`ScheduleRunnerPlugin`](https://docs.rs/bevy/latest/bevy/app/struct.ScheduleRunnerPlugin.html).

Example Usage:
```rust
use bevy::prelude::*;

fn main() {
    App::new()
        .add_plugins(MinimalPlugins)
        .add_systems(Update, my_system)
        .run();
}

fn my_system(counter: Local<u32>) {
    *counter += 1;
    println!("Counter: {}", *counter);
}
```

**Comment by Wandalen** [comment]
Awesome! That works ❤️
